src/phpDocumentor/UriFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor;
+
+use InvalidArgumentException;
+use League\Uri\Contracts\UriInterface;
+use League\Uri\Uri as LeagueUri;
+use Throwable;
+use const DIRECTORY_SEPARATOR;
+use function preg_match;
+use function sprintf;
+use function str_replace;
+use function strlen;
+use function strpos;
+use function substr;
+
+final class UriFactory
+{
+    public const WINDOWS_URI_FORMAT = '~^(file:\/\/\/)?(?<root>[a-zA-Z][:|\|])~';
```

Comment:
It feels weird to have this public and a value object consuming this; it makes for coupling between DSN and this service. Perhaps making this a proxy (as suggested earlier) with a helper method will resolve that?